### PR TITLE
feat: Added Financing object that can be referenced by Issuance trans…

### PR DESCRIPTION
**Pull request by lavens**
What type of PR is this?
/kind feature

What this PR does / why we need it:
This PR defines a Financing object and allows issuances to refer to it via the financing object's ID. Per issue https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/issues/445, this PR would assist in viewing all of the issuances created as part of a "financing".

Which issue(s) this PR fixes:
Fixes https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/issues/445

### CODE REVIEW

**Comment by JSv4** [comment]
Thanks @lavens. This looks great. Let's discuss on the call, but this works for me!

**Review by pjohnmeyer — COMMENTED** [review]
This makes sense to me @lavens, I just want to propose one alternative. Instead of each Issuance referencing a `financing_id`, would it be "better" to have the financing object reference an array of `resulting_security_ids`? Since this field is only interesting if there is a Financing object present, then it seems reasonable to me that we might stuff that information on the financing object.

On the other hand, your approach feels a little more compatible with an API that might be linking or unlinking individual instances.

**Comment by lavens** [comment]
> This makes sense to me @lavens, I just want to propose one alternative. Instead of each Issuance referencing a `financing_id`, would it be "better" to have the financing object reference an array of `resulting_security_ids`? Since this field is only interesting if there is a Financing object present, then it seems reasonable to me that we might stuff that information on the financing object.
> 
> On the other hand, your approach feels a little more compatible with an API that might be linking or unlinking individual issuances?

@pjohnmeyer That makes sense to me. This suggested approach would remove the need to update the issuance object and keep all relevant information isolated to the financing object. I'll make this change as a second commit in this PR so both options can be reviewed by the group.

**Review by pjohnmeyer — APPROVED** [review]
Per TWG Call: Agreed to this approach

File: schema/objects/Financing.schema.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      "type": "array",
+      "items": {
+        "type": "string",
+        "minItems": 1
+      }
+    },
+    "comments": {}
+  },
+  "additionalProperties": false,
+  "required": ["name", "issuance_ids"],
```
Comment (jacobyavis) [inline schema/objects/Financing.schema.json:32]: Should there be a `date` associated with each financing?

**Comment by jacobyavis** [comment]
As a clean-up item, we should update some of the markdown documentation regarding file types
e.g. Architecture.md mentions and lists "eight file types"

Also, we need to add the new file type to `UriLookupForFileType.json` so that the validator utils will consider this new file type.

File: schema/objects/Financing.schema.json (hunk 1 above)
Comment (jacobyavis) [reply schema/objects/Financing.schema.json:32]: The sample file mentions the date in the comments section
```
{
  "file_type": "OCF_FINANCINGS_FILE",
  "items": [
    {
      "object_type": "FINANCING",
      "id": "5bfe12c6-f049-4216-9f94-5a75dc3aaee5",
      "name": "Series A Funding",
      "issuance_ids": ["test-stock-issuance-minimal"],
      "comments": ["A funding round led by Firm Y on 02/15/2024."]
    }
  ]
}
```
In my opinion this data should be structured

**Comment by jacobyavis** [comment]
One more note: we missed adding FINANCING to the ObjectType schema, which was caught by the new validation I', wrking on for #470

https://github.com/Open-Cap-Table-Coalition/Open-Cap-Format-OCF/actions/runs/8837798105/job/24267465832?pr=491

**Comment by lavens** [comment]
@jacobyavis thanks for the review and suggestions. I did see that a few of the issues  are addressed in the validation PR #491 . That said I created a PR, to isolate the financing related changes, to address the found issues, update the generated schema docs, and include the proposed `date` property, #494

**Pull request merged**